waveorder_ros/waveorder_ros/phase_reconstruction_node.py

```diff
@@ -13,42 +13,57 @@ class PhaseReconstructionNode(Node):
     def __init__(self):
         super().__init__('phase_reconstruction_node')
         
+        self.setup_parameters()
+        
         self.get_logger().info("Generating phase calibration")
         # calculate the phase calibration
-        self.simulation_arguments = {
-            "zyx_shape": (11, 2448, 2048), # update to match data shape, probably (11, 2448, 2048)
-            "yx_pixel_size": 0.345, # in um, camera has 3.45 um pixels and we use 10x magnification
-            "z_pixel_size": 3, # in um, 1V piezo steps correspond to 3 um steps, update as needed
-            "index_of_refraction_media": 1, # this is an air immersion objective
-        }
-
-        self.transfer_function_arguments = {
-            "z_padding": 5, # it help to add padding to avoid wrapping artifacts at the edges
-            "wavelength_illumination": 0.532, # in um, green light
-            "numerical_aperture_illumination": 0.4, # this is a rough estimate
-            "numerical_aperture_detection": 0.45, # the objective has 0.45 NA detection
-        }
         
-        zyx=self.simulation_arguments["zyx_shape"]
-        self.image_z_stack = torch.empty(zyx)
+        self.image_z_stack = np.zeros( (self.get_parameter("zyx_shape").value[0],
+                                        self.get_parameter("zyx_shape").value[1] * self.get_parameter("zyx_shape").value[2]),
+                                        dtype=np.uint8)
         self.generate_phase_calibration()
 
         self.image_count = 0
         self.sub = self.create_subscription(Image, 'image_raw', self.image_callback, 11)
+        self.pub = self.create_publisher(Float32MultiArray, 'phase_reconstruction', 10)
+        
+        # Initialize the phase reconstruction volume message
+        self.phase_recon_msg = Float32MultiArray()
+        self.phase_recon_msg.layout.dim = [MultiArrayDimension(), MultiArrayDimension(), MultiArrayDimension()]
+        self.phase_recon_msg.layout.dim[0].label = "z"
+        self.phase_recon_msg.layout.dim[0].size = self.get_parameter("zyx_shape").value[0]
+        self.phase_recon_msg.layout.dim[0].stride = self.get_parameter("zyx_shape").value[1] * self.get_parameter("zyx_shape").value[2]
```

Comment:
Possible microöptimization: Is there much of a cost to `self.get_parameter("zyx_shape").value`? If so, you might cache it above this block of lines.